socks5.py

```diff
@@ -9,11 +10,28 @@
 import struct
 import threading
 from socketserver import ThreadingMixIn, TCPServer, StreamRequestHandler
+import concurrent.futures
+import time
+from datetime import datetime
+import sys
+import ipaddress
+if 'Pythonista' in sys.executable:
+    import console
+
+shutdown_event = threading.Event()
```

Comment:
Minor point: I would prefer to pass this `shutdown_event` object from the main thread to the traffic thread as an argument, rather than using a global. This is because Pythonista maintains a single global scope across all script runs, and simply deletes all the variables when the script is stopped. Using globals is hazardous because restarting the script soon after stopping it may result in the old thread seeing a new global event and failing to exit.